pkg/buildah/inspect.go

```diff
@@ -252,37 +248,34 @@ func inspectImage(ctx context.Context, sc *types.SystemContext, store storage.St
 		st.SetStore(store)
 	}
 
-	// try out the candidates as resolved by shortnames. This takes
-	// "localhost/" prefixed images into account as well.
-	candidates, err := shortnames.ResolveLocally(sc, imgName)
-	if err != nil {
-		return nil, nil, fmt.Errorf("error resolve shortname %s: %w", imgName, storage.ErrImageUnknown)
-	}
-
-	var errs []error
-	for _, candidate := range candidates {
-		f := func() (types.ImageSource, error) {
-			ref, err := transport.ParseReference(candidate.String())
-			if err != nil {
-				return nil, err
-			}
-			innerSrc, err := ref.NewImageSource(ctx, sc)
-			if err != nil {
-				return nil, err
-			}
-			return innerSrc, nil
+	parseSource := func(s string) (types.ImageSource, error) {
+		ref, err := transport.ParseReference(s)
+		if err != nil {
+			return nil, err
 		}
-		res, err := f()
+		return ref.NewImageSource(ctx, sc)
+	}
+	var (
+		src types.ImageSource
+		err error
+	)
+	switch transport.Name() {
+	case TransportContainersStorage:
+		names, err := shortnames.ResolveLocally(sc, imgName)
 		if err != nil {
-			errs = append(errs, err)
-		} else {
-			logger.Debug("using image name %q", candidate.String())
-			src = res
-			break
+			return nil, nil, fmt.Errorf("error resolve shortname %s: %w", imgName, storage.ErrImageUnknown)
+		}
+		for _, name := range names {
+			src, err = parseSource(name.String())
+			if err == nil {
+				break
+			}
```

Comment:
Initially, an `err` (we can call it err1) variable is declared via `var`, and this line `names, err := shortnames.ResolveLocally(sc, imgName)` (we can call it err2) is also declare an `err` variable, which is in the `case TransportContainersStorage:`. 

So, if `src, err = parseSource(name.String())` return an error, it will assign the value to err2, which will eventually cause the function to exit without returning an error and the `src` variable is still a nil pointer.

Reply:
I see, thanks for your kind interpretation. 👍